src/component/plotly-dash-preview/parse.js

```diff
@@ -0,0 +1,30 @@
+const isNonEmptyString = require('../../util/is-non-empty-string')
+
+/**
+ * @param {object} body : JSON-parsed request body
+ *  - url
+ *  - pdfOptions
+ * @param {object} opts : component options
+ * @param {function} sendToRenderer
+ * - errorCode
+ * - result
+ */
+function parse (body, opts, sendToRenderer) {
+  const result = {}
+
+  const errorOut = (code, msg) => {
+    result.msg = msg
+    sendToRenderer(code, result)
+  }
+
+  if (isNonEmptyString(body.url)) {
+    result.url = body.url
+  } else {
+    return errorOut(400, 'invalid url')
+  }
+
+  result.pdfOptions = body.pdf_options || {}
```

Comment:
What does Electron do when we pass invalid pdf options to `printToPDF`? 

- Does it ignore bad values and use the defaults?
- Does it error out?

if the latter, we should try to intercept bad pdf options here, so that we return a 400-series parse error instead of 500-series render error.

Reply:
... if the former, then we don't need to do anything here :tada:

Reply:
> Does it ignore bad values and use the defaults?

That's correct. No 500s ..

Reply:
Ok that's great. We won't have to worry about the various `printToPDF` options :ok_hand:

Reply:
It would be less frustrating to the user if we could return a 400-series error (and have that bubble up through streambed). For example if a user is trying to set the page size they may try `'page_size': 'legal'`, get back a letter option (because that's the default), and keep trying variations of `legal`. If we instead sent a 400 error saying that `page_size` is an invalid option, that will be clearer to the user.

If this is a lot of work, I don't feel strongly about it though.

Reply:
I agree. Returning a 400-series error here would be best from the user's standpoint. But since we would have to duplicate how Electron validates input here in the parse step in order to get this right, this may turn into a developer's nightmare for little gain. I'd vote for leaving this as is.

Reply:
OK, if it's easy let's do it but I agree that this isn't worth it if it's painful or a lot of work.